### PR TITLE
Add instruction for hyperparameter optimization using `allennlp-optuna`

### PR DESCRIPTION
**Pull request by himkt**
This PR adds an explanation to run hyperparameter optimization using [`allennlp-optuna`](https://github.com/himkt/allennlp-optuna). I think it would help users who want to optimize their model in allennlp cli.

### CODE REVIEW

**Comment by himkt** [comment]
Hello @epwalsh, I'd like some opinion or suggestion for the PR.
I'm sorry if the change looks like self-advertising (I don't have such intention).
If allennlp-guide has something policy that doesn't contain a personal project, I'd close it.

**Comment by epwalsh** [comment]
Hi @himkt, sorry we took a while to respond to this, I wasn't subscribed to notifications on this repo until now. This would be a good addition to the guide and we appreciate your contributions. I'll take a look at this on Monday :)

**Comment by himkt** [comment]
I'm relieved to hear from you. 😸  Thank you for reviewing, I will be waiting!

File: chapters/part2/hyperparameter-optimization.md [hunk 1]
```diff
@@ -345,6 +345,79 @@ if __name__ == '__main__':
...
+      "name": "lr",
+      "low": 5e-3,
+      "high": 5e-1,
+      "log": true
+    }
+  }
+]
+```
+
+After installing `allennlp-optuna`, you can launch optimization by the following command:
```
Comment (epwalsh) [inline chapters/part2/hyperparameter-optimization.md:405]:
```suggestion
You can then launch optimization with the following command:
```

File: chapters/part2/hyperparameter-optimization.md [hunk 2]
```diff
@@ -345,6 +345,79 @@ if __name__ == '__main__':
     )
 ```
 
+You can also use [`allennlp-optuna`](https://github.com/himkt/allennlp-optuna) for optimization.
+To use `allennlp-optuna`, you need to define a search space for hyperparameters in JSON.
```
Comment (epwalsh) [inline chapters/part2/hyperparameter-optimization.md:349]:
```suggestion
You can also use [`allennlp-optuna`](https://github.com/himkt/allennlp-optuna) for hyperparameter optimization.
After installing `allennlp-optuna`, you need to define a search space for hyperparameters in JSON:
```

File: chapters/part2/hyperparameter-optimization.md [hunk 3]
```diff
@@ -345,6 +345,79 @@ if __name__ == '__main__':
...
+    config/imdb_optuna.jsonnet # allennlp config file \
+    config/hparams.json # search space defined above  \
+    --serialization-dir result/optuna \
+    --study-name allennlp-optuna_demo \
+    --timeout 36000 \
+    --direction maximize
+```
+
+For more information about `allennlp-optuna` (including an installation instruction),
+please check [README](https://github.com/himkt/allennlp-optuna) of `allennlp-optuna` on GitHub.
```
Comment (epwalsh) [inline chapters/part2/hyperparameter-optimization.md:418]:
```suggestion
For more information about `allennlp-optuna` (including installation instructions),
please see the [README](https://github.com/himkt/allennlp-optuna) on GitHub.
```

**Review by epwalsh — COMMENTED** [review]
This looks great! Just a couple small suggestions for wording.

**Comment by himkt** [comment]
Thank you for taking a look! I applied your suggestions.

File: chapters/part2/hyperparameter-optimization.md [hunk 4]
```diff
@@ -345,6 +345,79 @@ if __name__ == '__main__':
...
+  }
+]
+```
+
+You can then launch optimization with the following command:
+
+```bash
+allennlp tune \
+    config/imdb_optuna.jsonnet # allennlp config file \
+    config/hparams.json # search space defined above  \
```
Comment (epwalsh) [inline chapters/part2/hyperparameter-optimization.md:410]:
```suggestion
    config/imdb_optuna.jsonnet \
    config/hparams.json \
```

**Comment by epwalsh** [comment]
Thanks @himkt!

**Pull request merged**